pkg/cidata/cidata.TEMPLATE.d/boot/30-install-packages.sh

```diff
@@ -23,149 +23,164 @@ if [ "${LIMA_CIDATA_UDP_DNS_LOCAL_PORT}" -ne 0 ] || [ "${LIMA_CIDATA_TCP_DNS_LOC
 fi
 
 # Install minimum dependencies
+# Run any user provided dependency scripts first
+if [ -d "${LIMA_CIDATA_MNT}"/provision.dependency ]; then
+	echo "Detected dependency provisioning scripts, running before default dependency installation"
+	for f in "${LIMA_CIDATA_MNT}"/provision.dependency/*; do
+		echo "Executing $f"
+		if ! "$f"; then
+			echo "Failed to execute $f"
```

Comment:
Is there a reason we don't want to `exit 1` here? The rest of the script runs under `set -e`, so any error return would immediately exit the script.

On the other hand, in `boot.sh`, we set `CODE=1` when a provisioning script fails, and then `exit $CODE` at the end.

Reply:
Makes sense. Changed it to collect all errors in a CODE var and exit if its 1 after all dependency scripts, to match boot.sh behavior